### ISSUE TITLE
[RFC] add fentry/fexit as mirrors of kfunc/kretfunc

### ISSUE THREAD

**Issue by jordalgo** [iovisor/bpftrace#2835]
### Is your feature request related to a problem? Please describe.
As @viktormalik informed me, the 'kfunc' name was in bpftrace first and was "borrowed" by the kernel to now mean something completely different (yet related to bpf). New (and perhaps even existing) users going to look for fentry/fexit equivalents in bpftrace are going to be confused (as I and Andrii both were) that they are called `kfunc` and `kretfunc` here.

### Describe the solution you'd like
Add `fentry` and `fexit` as mirrors for `kfunc` and `kretfunc` then update the documentation to favor `fentry` and `fexit` while still supporting `kfunc` and `kretfunc` (for as long as we have to).

### Describe alternative solutions or features you've considered
Living with it and crying.

**Comment by danobi** [comment]
I think aliasing `fentry`/`fexit` to  `kfunc`/`kretfunc` would make sense for the reasons you gave above. Documenting the alias would help for kernel folks that are grepping.

But FWIW, I'd also like to express that I think `kfunc`/`kretfunc` are good names. `kprobe` is a kernel probe that can be attached at any offset. `kfunc` attaches to a kernel function. It's internally consistent.

`fentry` is technically borrowed from gcc:

```
-mfentry
-mno-fentry

    If profiling is active (-pg), put the profiling counter call before the prologue. Note: On x86 architectures the attribute ms_hook_prologue isn’t possible at the moment for -mfentry and -pg.
```

So in other words it's not restricted to the kernel - userspace can also build with `fentry`. Maybe one day we get `ufunc` and `uretfunc`.

**Comment by viktormalik** [comment]
Agreed with @danobi here. I do like the `kfunc`/`kretfunc` name as it really nicely complements with `kprobe`/`kretprobe`. OTOH, users coming from other BPF tools may be be looking for `fentry`/`fexit` so creating the alias sounds reasonable.

**Comment by jordalgo** [comment]
Added here: https://github.com/iovisor/bpftrace/pull/2844
